### PR TITLE
feat(layer): introducing web component

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#10197 

### Description
This PR creates the new Layer component that is being used in many component story variations of the v11 Storybook. This PR also adds CDN and CodeSandbox examples, as well as documentation.

### Changelog

**New**

- new `cds-layer` component

### CODE REVIEW

**Comment by kennylam** [comment]
Minor thing but when going from `useLayer` back to `Default`, the component copy doesn't revert back.

image

**Comment by IgnacioBecerra** [comment]
@RichKummer That's a Storybook reference, the wrapper styles should've been updated [here](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10211) to match with Carbon's SB styles.

@kennylam You're right, fixing this!

File: packages/carbon-web-components/examples/codesandbox/basic/components/layer/cdn.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+<html>
+<head>
+  <title>@carbon/ibmdotcom-web-components example</title>
+  <meta charset="UTF-8"/>
+  <link rel="stylesheet"
+        href="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/tag/v1/latest/plex.css"/>
+  <style>
+    /* Suppress custom element until styles are loaded */
+    bx-button:not(:defined) {
```
Comment (kennylam) [inline packages/carbon-web-components/examples/codesandbox/basic/components/layer/cdn.html:18]:
```suggestion
    cds-button:not(:defined) {
```

File: packages/carbon-web-components/examples/codesandbox/basic/components/layer/index.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+<html>
+<head>
+  <title>carbon-web-components example</title>
+  <meta charset="UTF-8" />
+  <link rel="stylesheet"
+        href="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/tag/v1/latest/plex.css"/>
+  <style>
+    /* Suppress custom element until styles are loaded */
+    bx-button:not(:defined) {
```
Comment (kennylam) [inline packages/carbon-web-components/examples/codesandbox/basic/components/layer/index.html:18]:
```suggestion
    cds-button:not(:defined) {
```

File: packages/carbon-web-components/src/components/layer/layer.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import styles from './layer.scss';
+
+/**
+ * Basic layer
+ *
+ * @element cds-layer
+ * @slot children - The elements contained within the component.
+ */
+@customElement(`${prefix}-layer`)
+class BXLayer extends LitElement {
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/layer/layer.ts:22]:
```suggestion
class CDSLayer extends LitElement {
```

File: packages/carbon-web-components/src/components/layer/layer.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+   * A selector that selects a layer component.
+   */
+  static get eventUseLayer() {
+    return `${prefix}-use-layer`;
+  }
+
+  static styles = styles;
+}
+
+export default BXLayer;
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/layer/layer.ts:80]:
```suggestion
export default CDSLayer;
```

File: packages/carbon-web-components/src/components/layer/layer.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  @property({ type: Number, reflect: true })
+  level = 0;
+
+  @property()
+  layers;
+
+  updated() {
+    if (!this.layers) {
+      this.layers = this.querySelectorAll(
+        (this.constructor as typeof BXLayer).selectorLayer
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/layer/layer.ts:35]:
```suggestion
        (this.constructor as typeof CDSLayer).selectorLayer
```

File: packages/carbon-web-components/src/components/layer/layer.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+    this.layers.forEach((item) => {
+      (item as HTMLElement).setAttribute(
+        'level',
+        ((this.level + 1) % 3).toString()
+      );
+    });
+
+    this.dispatchEvent(
+      new CustomEvent((this.constructor as typeof BXLayer).eventUseLayer, {
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/layer/layer.ts:47]:
```suggestion
      new CustomEvent((this.constructor as typeof CDSLayer).eventUseLayer, {
```

**Review by kennylam — CHANGES_REQUESTED** [review]
Looks good @IgnacioBecerra! Just some minor changes for the prefix.

**Review by RichKummer — APPROVED** [review]

**Pull request merged**